new-lamassu-admin/src/pages/Dashboard/Alerts/Alerts.js

```diff
@@ -0,0 +1,122 @@
+import Button from '@material-ui/core/Button'
+import Grid from '@material-ui/core/Grid'
+import { makeStyles } from '@material-ui/core/styles'
+import React, { useState, useEffect } from 'react'
+
+import { Label1, H4 } from 'src/components/typography'
+
+import styles from '../Dashboard.styles'
+
+import AlertsTable from './AlertsTable'
+
+const NUM_TO_RENDER = 3
+
+const data = {
+  alerts: [
+    { text: 'alert 1' },
+    { text: 'alert 2' },
+    { text: 'alert 3' },
+    { text: 'alert 4' },
+    { text: 'alert 5' }
+  ]
+}
+
+const useStyles = makeStyles(styles)
+
+const Alerts = ({ cardState, setRightSideState }) => {
+  const classes = useStyles()
+  const [showAllItems, setShowAllItems] = useState(false)
+  const [showExpandButton, setShowExpandButton] = useState(false)
+  const [numToRender, setNumToRender] = useState(NUM_TO_RENDER)
+
+  useEffect(() => {
+    if (showAllItems) {
+      setShowExpandButton(false)
+      setNumToRender(data?.alerts.length)
+    } else if (data && data?.alerts.length > numToRender) {
+      setShowExpandButton(true)
+    }
+    if (cardState.cardSize === 'small' || cardState.cardSize === 'default') {
+      setShowAllItems(false)
+      setNumToRender(NUM_TO_RENDER)
+    }
+  }, [cardState.cardSize, numToRender, showAllItems])
```

Comment:
Maybe you should move cardState.cardSize into a new useEffect since the if else if block it's not related to it, something like this:

`useEffect(() => {
    if (cardState.cardSize === 'small' || cardState.cardSize === 'default') {
      setShowAllItems(false)
      setNumToRender(NUM_TO_RENDER)
    }
  }, [cardState.cardSize]`